src/components/NavigationBar.jsx

```diff
@@ -0,0 +1,157 @@
+        <div className="logo_items flex">
+          <span className="nav_image">
+            <i className="bx bxl-airbnb bx-lg" color="orange" />
+          </span>
+          <span className="logo_name">AirBnB</span>
+          {isHovering && (
+          <button type="button" aria-label="Unlock Sidebar" className={`bx ${isLocked ? ' bx-md bx-lock-open-alt' : 'bx-md bx-lock-alt'} no-style-button`} onClick={toggleLock} title="Unlock Sidebar" />
+          )}
+          <button type="button" aria-label="Close Sidebar" className="no-style-button bx bx-x bx-md" onClick={toggleSidebar} />
+        </div>
+
+        <div className="menu_container">
+          <div className="menu_items">
+            <ul className="menu_item">
+              <li className="item">
+                <Nav.Link className="link flex capital-text" as={NavLink} to="/" onClick={toggleSidebar}>
+                  <i className="bx bx-sm bxs-building-house" />
+                  {' '}
+                  PLACES
+                </Nav.Link>
+              </li>
+              <li className="item">
+                <Nav.Link className="link flex capital-text" as={NavLink} to="/reserve" onClick={toggleSidebar}>
+                  <i className="bx bx-sm bx-calendar-check bx-tada" />
+                  {' '}
+                  RESERVE
+                </Nav.Link>
+              </li>
+              <li className="item">
+                <Nav.Link className="link flex capital-text" as={NavLink} to="/myreservations" onClick={toggleSidebar}>
+                  <i className="bx bx-sm bx-list-check" />
+                  {' '}
+                  MY RESERVATIONS
+                </Nav.Link>
+              </li>
+              <li className="item">
+                <Nav.Link className="link flex capital-text" as={NavLink} to="/deleteplace" onClick={toggleSidebar}>
+                  <i className="bx bx-sm bx-bookmark-minus" />
+                  {' '}
+                  DELETE PLACE
+                </Nav.Link>
+              </li>
+              <li className="item">
+                <Nav.Link className="link flex capital-text" as={NavLink} to="/addplace" onClick={toggleSidebar}>
+                  <i className="bx bx-sm bx-location-plus" />
+                  {' '}
+                  ADD PLACE
+                </Nav.Link>
+              </li>
+              <li className="item">
+                {userId ? (
+                  <>
+                    <Nav.Link className="link flex capital-text" href="/">
+                      <i className="bx bx-sm bx-log-out" />
+                      {' '}
+                      LOG OUT
+                    </Nav.Link>
+                  </>
+                ) : (
+                  <Nav.Link className="link flex capital-text" as={NavLink} to="/login" onClick={toggleSidebar}>
+                    <i className="bx bx-sm bx-log-in" />
+                    {' '}
+                    LOG IN
+                  </Nav.Link>
```

Comment:
[OPTIONAL]

There may be unnecessary repetition here. These links seem to be similar, maybe the code can be condensed with some sort of looping, and array

OR abstraction through a smaller component like so:


```js
const NavigationBar = () => {
  const [sidebarState, setSidebarState] = useState({
    isLocked: true,
    isHoverable: true,
    isClosed: true,
    isHovering: false,
    isNavbarVisible: true,
  });

.....

  return (
    // ... remains the same, replace Nav.Link items with SidebarLink
    <SidebarLink to="/" icon="bxs-building-house" label="PLACES" onClick={toggleSidebar} />
    // ...
  );

```

or BOTH.